src/deluge/gui/views/performance_session_view.cpp

```diff
@@ -0,0 +1,2045 @@
+	char const* tagName;
+	while (*(tagName = storageManager.readNextTagOrAttributeName())) {
+		//<param>
+		if (!strcmp(tagName, STRING_FOR_PERFORM_DEFAULTS_PARAM_TAG)) {
+			readDefaultFXParamFromFile(xDisplay);
+		}
+		//<row>
+		else if (!strcmp(tagName, STRING_FOR_PERFORM_DEFAULTS_ROW_TAG)) {
+			readDefaultFXRowNumberValuesFromFile(xDisplay);
+		}
+		//<hold>
+		else if (!strcmp(tagName, STRING_FOR_PERFORM_DEFAULTS_HOLD_TAG)) {
+			readDefaultFXHoldStatusFromFile(xDisplay);
+		}
+		storageManager.exitTag();
+	}
+}
+
+//compares param name from <param> tag to the list of params available for use in performance view
+//if param is found, it loads the layout info for that param into the view (paramKind, paramID, xDisplay, yDisplay, rowColour, rowTailColour)
+void PerformanceSessionView::readDefaultFXParamFromFile(int32_t xDisplay) {
+	char const* paramName;
+	char const* tagName = storageManager.readTagOrAttributeValue();
+
+	for (int32_t i = 0; i < kNumParamsForPerformance; i++) {
+		if (songParamsForPerformance[i].paramKind == Param::Kind::UNPATCHED_GLOBAL) {
+			paramName = GlobalEffectable::paramToString(Param::Unpatched::START + songParamsForPerformance[i].paramID);
+		}
+		else if (songParamsForPerformance[i].paramKind == Param::Kind::UNPATCHED_SOUND) {
+			paramName =
+			    ModControllableAudio::paramToString(Param::Unpatched::START + songParamsForPerformance[i].paramID);
+		}
+		if (!strcmp(tagName, paramName)) {
+			memcpy(&layoutForPerformance[xDisplay], &songParamsForPerformance[i], sizeParamsForPerformance);
+
+			memcpy(&backupXMLDefaultLayoutForPerformance[xDisplay], &layoutForPerformance[xDisplay],
+			       sizeParamsForPerformance);
+			break;
+		}
+	}
+}
+
+void PerformanceSessionView::readDefaultFXRowNumberValuesFromFile(int32_t xDisplay) {
+	char const* tagName;
+	char rowNumber[5];
+	//loop through all row <#> number tags
+	while (*(tagName = storageManager.readNextTagOrAttributeName())) {
+		//find the row number that the tag corresponds to
+		//reads from row 8 down to row 1
+		for (int32_t yDisplay = kDisplayHeight - 1; yDisplay >= 0; yDisplay--) {
+			intToString(yDisplay + 1, rowNumber);
+			if (!strcmp(tagName, rowNumber)) {
+				defaultFXValues[xDisplay][yDisplay] = storageManager.readTagOrAttributeValueInt() - kKnobPosOffset;
+
+				//check if a value greater than 64 was entered as a default value in xml file
+				if (defaultFXValues[xDisplay][yDisplay] > kKnobPosOffset) {
+					defaultFXValues[xDisplay][yDisplay] = kKnobPosOffset;
+				}
+
+				backupXMLDefaultFXValues[xDisplay][yDisplay] = defaultFXValues[xDisplay][yDisplay];
+
+				break;
+			}
+		}
+		storageManager.exitTag();
+	}
+}
+
+void PerformanceSessionView::readDefaultFXHoldStatusFromFile(int32_t xDisplay) {
+	char const* tagName;
+	//loop through the hold tags
+	while (*(tagName = storageManager.readNextTagOrAttributeName())) {
+		//<status>
+		if (!strcmp(tagName, STRING_FOR_PERFORM_DEFAULTS_HOLD_STATUS_TAG)) {
```

Comment:
I don't think we want to use the `STRING_FOR_*` constants here, instead the prevailing style seems to be just having the tag name directly. This would matter especially if we release translations.

Reply:
ah I see. Maybe prefix them with `TAG_FOR_` instead to make it clearer that these aren't localization strings?

Reply:
> ah I see. Maybe prefix them with `TAG_FOR_` instead to make it clearer that these aren't localization strings?

I just removed the "STRING_FOR" prefix altogether. Should be good now :)!